src/main/java/net/dv8tion/jda/api/utils/messages/MessageEditBuilder.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.api.utils.messages;
+
+import net.dv8tion.jda.api.entities.IMentionable;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.components.LayoutComponent;
+import net.dv8tion.jda.api.utils.AttachedFile;
+import net.dv8tion.jda.api.utils.FileUpload;
+import net.dv8tion.jda.internal.utils.AllowedMentionsImpl;
+import net.dv8tion.jda.internal.utils.Checks;
+import net.dv8tion.jda.internal.utils.Helpers;
+import net.dv8tion.jda.internal.utils.IOUtil;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Builder specialized for building a {@link MessageEditData}.
+ *
+ * <p>These are used to edit messages and allow configuration that either {@link #replace(boolean) replaces} the message or only updates specific fields.
+ *
+ * @see MessageCreateBuilder
+ */
+@SuppressWarnings("ResultOfMethodCallIgnored")
+public class MessageEditBuilder extends AbstractMessageBuilder<MessageEditData, MessageEditBuilder> implements MessageEditRequest<MessageEditBuilder>
+{
+    protected static final int CONTENT       = 1;
+    protected static final int EMBEDS        = 1 << 1;
+    protected static final int COMPONENTS    = 1 << 2;
+    protected static final int ATTACHMENTS   = 1 << 3;
+    protected static final int MENTIONS      = 1 << 4;
+
+    protected static final int ALL = CONTENT | EMBEDS | COMPONENTS | ATTACHMENTS | MENTIONS;
+
+    private boolean replace = false;
+    private int set = 0;
+
+    private final List<AttachedFile> attachments = new ArrayList<>(10);
```

Comment:
If this 10 comes from a known constant, lets use that.

Reply:
The API actually doesn't limit the number of files, 10 is just a reasonable expectation.